ex4/api-server/tasks.js

```diff
@@ -0,0 +1,78 @@
+const express = require('express');
+const fs = require('fs');
+
+// create an instance of express to serve our end points
+const app = express();
+const path = './Tasks.json'
+
+// create a const variable for the port number
+const PORT = 3000;
+
+// read tasks list
+app.get('/tasks', (req, res) => {
+    console.log('Got request to /tasks')
+    fs.readFile(path, (err, data) => {
+        if (err) {
+            throw err;
+        }
+        res.send(JSON.parse(data));
+    });
+});
+
+const readFile = (callback, filePath = path) => {
+    fs.readFile(filePath, (err, data) => {
+        if (err) {
+            throw err;
+        }
+
+        callback(JSON.parse(data));
+    });
+};
+
+const writeFile = (fileData, callback, filePath = path) => {
+
+    fs.writeFile(filePath, fileData, (err) => {
+        if (err) {
+            throw err;
+        }
+
+        callback();
+    });
+};
+
+// create new task
+app.post('/tasks/new', (req, res) => {
```

Comment:
the /new is redundant.
restful api should be using a single controller with the different method types e.g:
GET /tasks should return all tasks
GET /tasks?id=1 should return task with id 1
POST /tasks should insert a new task
DELETE /tasks?id=1 should delete task with id 1
etc..